README.md

```diff
@@ -1,5 +1,98 @@
+
+## Sequence diagrams
+
+### Fund executed
+
+The following diagram shows the sequence of the payout creation and execution.
+
+![Fundable Token: Fund executed](http://www.plantuml.com/plantuml/png/bP0nhi8m44HxdsALFXVunLSG0eag57004qyQmZhQNKiS7pkcGWH4PutVQ9wsK_DYwJQviq8srFvJiqyS_6K2dU8LtFnHb00fgNlzexOKbZTj8vlumQYV8TwYv6R1-z96SAlO-z5PuW3L3xEByNFamCVAkoQLcCOSwbU_7B2Ea8h_geWxXcHOLrmCg-2kii5ru93l_G00)
+
+### Fund cancelled
+
+The following diagram shows the sequence of the payout creation and cancellation.
+
+![Fundable Token: Fund cancelled](http://www.plantuml.com/plantuml/png/SoWkIImgAStDuGejJYroLD2rKr1oAyrBIKpAILK8oSzEpLEoKiWlIaaj0eboeSifw88qWDaAplbv9KM9oIKA_WMfHOab-KL0dI3b_Y3TgIbS-6n8JinBJiqXsmfdiFOSe2KEgNafGDy00000)
+
+### Fund rejected
+
+The following diagram shows the sequence of the payout creation and rejection.
+
+![Fundable Token: Fund rejected](http://www.plantuml.com/plantuml/png/SoWkIImgAStDuGejJYroLD2rKr1oAyrBIKpAILK8oSzEpLEoKiWlIaaj0eboeSifw88qWDaAplbv9KM9oIKA_WMfHOab-KL0dI3b_Y3TgIbSN3ZLORUrA3KhDRa4ZxLW5qCj2G8h1jhYa9gN0lGE0000)
+
+## State diagram
+
+![Fundable: State Diagram](http://www.plantuml.com/plantuml/png/TP71JiGm34Jl-Oev8_O7EA2L44YSG6XlY4EQk518xLNY2FhtjDscAK8zLMDFvzdECQcDcljWNg_US3ZSuYbwYkHly93PBy8itzdmoCzH7ALqiQpIh7-Uv8iumKR2cTXt_0uLDAEwECcZ6Q0V19sSSC8QhLF8cQ9LYMR3E8ssTkGipKuOtpPXdflyDPgi9Qf0MHG3AJgUEuqDbXr1sbdcH1l0lDBz3xciPUcYJ2sV_WBwZiSoMearSNkYkuaNGWbSfQcDoGFXCA0XE1j2FSg_k1Vw9BUELzHsXtrJBUy3zrxXIEnp7tu1)
+
+## Install
+
+```
+npm install eip2019
+```
+
+## Usage
+
+To write your custom contracts, import the contract and extend it through inheritance.
+
+```solidity
+pragma solidity ^0.5.0;
+
+import 'eip2019/contracts/Fundable.sol';
+
+contract MyFundable is Fundable {
+    // your custom code
+}
+```
+
+> You need an ethereum development framework for the above import statements to work! Check out these guides for [Truffle], [Embark] or [Buidler].
+
+## Fund information
+
+Whenever a payout is ordered, payment information has to be provided with the necessary information for the off-chain transfer. [EIP-2019](https://github.com/ethereum/EIPs/pull/2019/files) leaves the structure of this information up to the implementer, but recommends [ISO-20022](https://en.wikipedia.org/wiki/ISO_20022) as a starting point.
+
+The unit tests use a JSON version of this standard, which can be seem below.
+
+```json
+{
+    "messageId": "Example Message ID",
+    "funds": [
+        {
+            "amount": 1.00,
+            "fundSourceId": "caaa2bd3-dc42-436a-b70b-d1d7dac23741",
+            "receiverInformation": "Example funds receiver information"
```

Comment:
I think this must me named fundingSubject, mainly you are funding your own wallet

Reply:
I've changed it now

Reply:
I meant the receivinginformation as fundingSubject  :) sorry